x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java

```diff
@@ -265,7 +265,12 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
         }
 
         void run() {
-            writeRandomBlob(request.readEarly || random.nextBoolean(), true, this::doReadBeforeWriteComplete, write1Step);
+            writeRandomBlob(
+                request.readEarly || (request.targetLength <= Integer.MAX_VALUE && random.nextBoolean()),
```

Comment:
nit: is it possible to be a little more explicit here and use `ByteSizeValue.ofGb(2)`?

Reply:
IMO the right way to describe the bound is `Integer.MAX_VALUE` - the reason for this limit is because the API for atomic writes forces the use of `int` rather than `long` in some places and we must avoid overflow in that conversion. It happens to be the same as `ByteSizeValue.ofGb(2)` (minus 1) but if anything that's the coincidence.

Reply:
It's an interesting comment tho, we could be more explicit. I introduced a constant for this limit so that we can record why the limit exists in 96008bf067678a7548d136564a540941c5ed22c9.

Reply:
Ah right, I missed that part. Thanks for adding the constant.